lib/usd/translators/shading/usdStandardSurfaceWriter.cpp

```diff
@@ -155,18 +162,28 @@ void PxrUsdTranslators_StandardSurfaceWriter::Write(const UsdTimeCode& usdTime)
         PxrMayaUsdPreviewSurfaceTokens->ClearcoatRoughnessAttrName,
         usdTime);
 
-    MPlug transmissionPlug = depNodeFn.findPlug(
-        depNodeFn.attribute(TrMayaTokens->transmission.GetText()),
+    MPlug opacityPlug = depNodeFn.findPlug(
+        depNodeFn.attribute(TrMayaTokens->opacity.GetText()),
         /* wantNetworkedPlug = */ true,
         &status);
-    if (status == MS::kSuccess && UsdMayaUtil::IsAuthored(transmissionPlug)) {
-        // Need a solution if the transmission is textured, but in the
-        // meantime, we go 1 - transmission.
-        float transmissionValue = transmissionPlug.asFloat();
 
-        shaderSchema
-            .CreateInput(PxrMayaUsdPreviewSurfaceTokens->OpacityAttrName, SdfValueTypeNames->Float)
-            .Set(1.0f - transmissionValue, usdTime);
+    if (UsdMayaUtil::IsAuthored(opacityPlug) || opacityPlug.numConnectedChildren()
+        || opacityPlug.isDestination()) {
+        UsdShadeInput opacityInput = shaderSchema.CreateInput(
+            PxrMayaUsdPreviewSurfaceTokens->OpacityAttrName, SdfValueTypeNames->Float);
+
+        if (!opacityPlug.numConnectedChildren()) {
+            VtValue opacityValue = UsdMayaWriteUtil::GetVtValue(
+                opacityPlug,
+                SdfValueTypeNames->Color3f,
+                /* linearizeColors = */ false);
+
+            // Need the luminance since we have a single float to populate on the USD side.
+            // This is the ACEScg luminance formula. Should we infer lin_rec709 instead?
```

Comment:
I was actually wondering about a similar thing - should this query color space from Maya?

Reply:
Added an OCIO v2 todo item.